### ISSUE TITLE
Build for Ubuntu Xenial Xerus 16.04

### ISSUE THREAD

**Issue by nmabhinandan** [agronick/Relay#40]

**Comment by kshenoy** [comment]
Seconded

**Comment by macbroadcast** [comment]
+1

**Comment by daviip** [comment]
Agreed. I tried building on 16.04, (although I am actually running eos Loki), but ran into errors when trying to run compile. Tracked down the issue to the gtk3 vapi file incorrectly converting to C code. Apparently, gtk 3.18 is too old for vala 32.1, and the minimum gtk version is 3.20. Anyway, made a patch for the bug that can be used to build on your own system. Just clone the repo, and download this [gtk-button.patch](https://github.com/daviip/Relay/blob/master/src/gtk-button.patch) and place it in the /src directory of the repo, then run these commands (from top level of repo):

`mkdir build &&  cd build`
`../autogen.sh`
`make`
`cd ../src && patch < gtk-button.patch`
`cd ../build`
`make`

Make will fail the first time, but second time should work (with a lot of warnings). Then you can just 'sudo make install' to install it. It seemed buggy to me though.

First time ever really using git. Hope I'm doing it right.

**Comment by kshenoy** [comment]
@daviip Thanks for responding, will give this a go.

> First time ever really using git. Hope I'm doing it right.

I suppose you used backticks to demarcate the lines with code. It might be easier to just add 4 spaces at the beginning of the line to indicate it's code or add a line with 3 backticks before and after the code block. Eg.

```
This is a code block
```

Check out other styling options by clicking on the _Styling with Markdown is supported_ link located below the edit box.

**Comment by sysfu** [comment]
I was able to get a build started on Loki after installing the following packages:
`automake autoconf libtool-bin libunity-dev`

...however it bails out with this error:

```
/usr/include/glib-2.0/glib/gstrfuncs.h:175:23: note: expected ‘gchar * {aka char *}’ but argument is of type ‘const gchar * {aka const char *}’
 gchar*                g_strchug        (gchar        *string);
                       ^
  CCLD     relay
main_window.o: In function `main_window_construct':
/usr/local/src/relay/build/src/../../src/main_window.c:1589: undefined reference to `gtk_widget_set_focus_on_click'
collect2: error: ld returned 1 exit status
Makefile:495: recipe for target 'relay' failed
make[2]: *** [relay] Error 1
make[2]: Leaving directory '/usr/local/src/relay/build/src'
Makefile:462: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/usr/local/src/relay/build'
Makefile:373: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by sysfu** [comment]
Upon further investigation, the offending line 1589 in main_window.c uses 'gtk_widget_set_focus_on_click'
...and it appears that this function was [not introduced until lib-gtk 3.2 ](url=gtk_widget_set_focus_on_click)

The version that ships with Loki is 3.18: ii  libgtk-3-dev:amd64                             3.18.9-1ubuntu3.1 

So the solution appears to be either edit main_window.c and roll back the function name to 'gtk_button_set_focus_on_click' or build a version of libgt-3-dev that is v3.20 or newer.

**Comment by sysfu** [comment]
I went with editing main_window.c and was able to successfully build the app from source and install.

**Comment by kshenoy** [comment]
> Upon further investigation, the offending line 1589 in main_window.c uses 'gtk_widget_set_focus_on_click'
...and it appears that this function was not introduced until lib-gtk 3.2
> 
> The version of libgtk-3-dev that ships with Loki is 3.18:

Wait! What? An older version of Elementary OS ships with a newer version of lib-gtk?

**Comment by sysfu** [comment]
My apologies, I left a zero off the version number, it should have been '3.20'. I have fixed it in the comment above as well as the link.

**Comment by mauritslamers** [comment]
I would also love to see relay for Loki. Is there anything any of us could do to help out?

**Comment by agronick** [comment]
You could just make a PPA. All the sources are there.

**Comment by yerbestpal** [comment]
I've never created a PPA but I suppose I could give it a bash.
